### PR TITLE
Clip small negative values in van_rossum_distance before np.sqrt

### PR DESCRIPTION
**Pull request by CozySocksAlways**
This PR fixes Issue #679 . Currently `van_rossum_distance` could return `nan` due to extremely small negative values because of  floating point errors before applying `np.sqrt`.

**Fix**
Added check before `np.sqrt` to find small negative values, then replaced them using `np.maximum(vr_dist,0)`. Also outputs runtime warning for transparancy.
This ensures that spike trains that should result in a distance of 0 no longer produce `nan`.

**Test**
Added unit test that reproduces the spike train scenario from the issue, it asserts that the computed distance is 0 and not `nan`

### CODE REVIEW

File: elephant/test/test_spike_train_dissimilarity.py [hunk 1]
```diff
@@ -599,7 +600,15 @@ def test_van_rossum_distance(self):
         self.assertEqual(stds.van_rossum_distance(
             [self.st21], self.tau3)[0, 0], 0)
         self.assertEqual(len(stds.van_rossum_distance([], self.tau3)), 0)
-
+        
+        # Check small negative values edge case
+        with warnings.catch_warnings(record='True') as w:
```
Comment (kohlerca) [inline elephant/test/test_spike_train_dissimilarity.py]: The expected type for argument `record` is `bool`, not `str`

File: elephant/test/test_spike_train_dissimilarity.py [hunk 2]
```diff
@@ -47,6 +47,7 @@ def setUp(self):
         self.st23 = StationaryPoissonProcess(rate=30 * Hz, t_start=0 * ms,
                                              t_stop=1000 * ms
                                              ).generate_spiketrain()
+        self.st24 = SpikeTrain([0.1782, 0.2286, 0.2804, 0.4972, 0.5504], units='s',t_stop=4.0)
```
Comment (kohlerca) [inline elephant/test/test_spike_train_dissimilarity.py]: As suggested by the dedicated regression test, the spike train can be initialized locally in the function, which facilitates code reading.

File: elephant/spike_train_dissimilarity.py [hunk 3]
```diff
@@ -363,6 +363,15 @@ def van_rossum_distance(spiketrains, time_constant=1.0 * pq.s, sort=True):
     for i, j in np.ndindex(k_dist.shape):
         vr_dist[i, j] = (
             k_dist[i, i] + k_dist[j, j] - k_dist[i, j] - k_dist[j, i])
+    
+    # Clip small negative values
+    if np.any(vr_dist < 0):
+        warnings.warn(
+                "van_rossum_distance: very small negative values encountered "
```
Comment (kohlerca) [inline elephant/spike_train_dissimilarity.py]: Here, it would also be helpful to hint at possible causes. As we tested, this does not happen if the units are in ms, and the user could change the representation to prevent the warning if noticed. Maybe something like "... very small negative values encountered, setting them to zero. This is likely due to floating point error, which can happen if the spike times are small floating point values. In this case, a possible solution to prevent the warning is to change the time unit of the input spike trains for better precision (e.g., from seconds to milliseconds).".

File: elephant/spike_train_dissimilarity.py [hunk 4]
```diff
@@ -363,6 +363,18 @@ def van_rossum_distance(spiketrains, time_constant=1.0 * pq.s, sort=True):
...
+
+    # Clip small negative values
+    if np.any(vr_dist < 0):
+        warnings.warn(
+            "van_rossum_distance: very small negative values encountered; "
+            "setting them to zero. Potentially due to floating point error, "
+            "which can occur if spike times are represented as small floating "
+            "point values (e.g., in seconds). A possible way to prevent this "
+            "warning is to use a time unit with better numerical precision, "
+            "e.g. seconds to milliseconds.", RuntimeWarning)
```
Comment (kohlerca) [inline elephant/spike_train_dissimilarity.py]: The revised warning looks good. I have a small suggestion for improved grammar `e.g., from seconds` instead of `e.g. seconds`
